src/include/daos_srv/srv_csum.h

```diff
@@ -50,4 +31,158 @@ ds_csum_add2iod(daos_iod_t *iod, struct daos_csummer *csummer,
 		struct bio_sglist *bsgl, struct dcs_csum_info *biov_csums,
 		size_t *biov_csums_used, struct dcs_iod_csums *iod_csums);
 
+
+/**
+ * The following declarations are for checksum scrubbing functions. The function
+ * types provide an interface for injecting dependencies into the
+ * scrubber (srv_pool_scrub.c) from the schedule/ult management
+ * so that it can be more easily tested without depending on the entire daos
+ * engine to be running or waiting for schedules to run.
+ */
+
+struct cont_scrub {
+	struct daos_csummer	*scs_cont_csummer;
+	daos_handle_t		 scs_cont_hdl;
+	uuid_t			 scs_cont_uuid;
+};
+
+/*
+ * Because the scrubber operates at the pool level, it will need a way to
+ * get some info for each container within the pool as it's scrubbed.
+ */
+typedef int(*ds_get_cont_fn_t)(uuid_t pool_uuid, uuid_t cont_uuid, void *arg,
+			    struct cont_scrub *cont);
+
+/*
+ * handler for scrubber progress. will get called after each checksum is
+ * calculated.
+ */
+typedef int (*ds_progress_handler_t)(void *);
+/* handler for when corruption is discovered */
+typedef int (*ds_corruption_handler_t)(void *);
+/* Inject schedule sleeps or yields */
+typedef int (*ds_sleep_fn_t)(void *, uint32_t msec);
+typedef int (*ds_yield_fn_t)(void *);
+
+/* For the pool target to start and stop the scrubbing ult */
+int ds_start_scrubbing_ult(struct ds_pool_child *child);
+void ds_stop_scrubbing_ult(struct ds_pool_child *child);
+
+enum scrub_status {
+	SCRUB_STATUS_UNKNOWN = 0,
+	SCRUB_STATUS_RUNNING = 1,
+	SCRUB_STATUS_NOT_RUNNING = 2,
+};
+
+/* telemetry metrics for the scrubber */
+#define DF_POOL_DIR "scrub/pool:"DF_UUID"[%d]"
+#define DP_POOL_DIR(ctx) DP_UUID((ctx)->sc_pool_uuid), \
+			(ctx)->sc_dmi->dmi_tgt_id
+
+#define DF_CONT_DIR DF_POOL_DIR "/cont:"DF_UUID
+#define DP_CONT_DIR(ctx) DP_POOL_DIR(ctx) , DP_UUID(ctx->sc_cont.scs_cont_uuid)
```

Comment:
(style)  space prohibited before that ',' (ctx:WxW)
(style)  Macros with complex values should be enclosed in parentheses